torch/csrc/jit/ir/alias_analysis.cpp

```diff
@@ -126,10 +126,78 @@ bool AliasDb::isContainerType(const TypePtr& type) const {
 
 AliasDb::~AliasDb() = default;
 
+// Structure that keeps tracks of all writes at a high level. When alias
+// analysis is done, this will be used to construct a more efficient WriteIndex.
+struct AliasDb::WriteRegistry {
+    void registerWrite(const Value* v, Node* n) {
+      writes_.emplace(n, v);
+    }
+    void registerWriteToAllContained(const Value* v, Node* n) {
+      containedWrites_.emplace(n, v);
+    }
+    void registerWriteToAllWildcards(Node* n) {
+      writesToAllWildcards_.insert(n);
+    }
+    std::unordered_map<Node*, const Value*> writes_;
```

Comment:
yes good catch, this is a bug. I'll fix. 

containedWrites are separate because it produces more restrictive results (e.g. appending to a list does not write to elements in the list, but it writes to list itself)